log.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef CNT4007_LOG_HPP
+#define CNT4007_LOG_HPP
+#include "PeerController.hpp"
+#include "PeerProcess.hpp"
+#include <ctime>
+
+class Log {
+    public:
+        Log(std::string path);
+	    std::ofstream& getStream();
+
+    private:     
+        ofstream logMsg;
+        string path;
+        void TCPconnection();
+        void changePrefNeighbors();
+        void changeUnchokeNeighbors();
+        void unchoking();
+        void choking();
+        void haveMsg();
+        void interestMsg();
+        void notInterestMsg();
+        void downloadPiece();
+        void downloadComplete();
+
+        // current date and time on the current system
+        time_t now = time(0);

```

Comment:
You need to get the timestamp for every function call. If you only call this once, the timestamp will never update.